Models/Review_models.js

```diff
@@ -11,13 +12,40 @@ function getReviewByIdModel(id) {
     })
 }
 
-function getReviewsModel() {
-    return db.query(`
-    SELECT reviews.owner, reviews.title, reviews.review_id, reviews.category, reviews.review_img_url, reviews.created_at, reviews.votes, reviews.designer, COUNT(comments.review_id) AS comment_count FROM reviews
-    FULL OUTER JOIN comments ON comments.review_id = reviews.review_id
-    GROUP BY reviews.review_id
-    ORDER BY reviews.created_at DESC;`)
+function getReviewsModel(category, sort_by, order) {
+    const queryValues = [];
+    let queryStr = `SELECT reviews.owner, reviews.title, reviews.review_id, reviews.category, reviews.review_img_url, reviews.created_at, reviews.votes, reviews.designer, COUNT(comments.review_id) AS comment_count FROM reviews FULL OUTER JOIN comments ON comments.review_id = reviews.review_id`
+    if (category){
+        queryValues.push(category)
+        queryStr += ` WHERE reviews.category = $1`
+    }
+    if (sort_by){
+        if (!['review_id', 'title', 'category', 'designer', 'owner', 'review_img_url', 'created_at', 'votes'].includes(sort_by)){
+        return Promise.reject({status: 400, msg: "Invalid sort query"})
+    }
+}
+    if (order) {
+        if (!['ASC', 'DESC'].includes(order)){
+        return Promise.reject({status: 400, msg: "Invalid order query"})
+    }
+}
+    if (sort_by && order){
+        queryStr += ` GROUP BY reviews.review_id ORDER BY reviews.${sort_by} ${order};`
+    }
+    if (sort_by && !order){
+        queryStr += ` GROUP BY reviews.review_id ORDER BY reviews.${sort_by} DESC;`
+    }
+    if (!sort_by && order){
+        queryStr += ` GROUP BY reviews.review_id ORDER BY reviews.created_at ${order};`
+    }
+    if (!sort_by && !order){
+        queryStr += ` GROUP BY reviews.review_id ORDER BY reviews.created_at DESC;`
```

Comment:
We may be able to refactor this to make it a little more DRY. Consider the order of these query strings then you could probably just worry about what there 'is' rather than what there 'isn't'. I.e. no if statements with '!' in.